### PR TITLE
Graduate extensions: Add deprecation notice banners

### LINKED ISSUE #38624: Add deprecation warning on extensions page

### LINKED ISSUE #39036: Add deprecation notice to the authoring extension page

### LINKED ISSUE #39037: Add deprecation notice to extension registry

### LINKED ISSUE #39039: Add deprecation notice Sourcegraph extensions architecture page
This [page](https://docs.sourcegraph.com/dev/background-information/architecture/sourcegraph-extensions)

### LINKED ISSUE #39040: Create notice on the extensions page that extensions administration works only on versions prior to 4.0

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #38624
Closes #39036
Closes #39037
Closes #39039
Closes #39040

A couple of things going on there:

- I found most of the doc pages that mention extensions and extension APIs and added a placeholder text. The styling follows the existing NOTE styling as seen here: https://docs.sourcegraph.com/admin/external_services/postgres
- I added a deprecation notice banner to all extensions pages on the web UI (screenshots in test plan)
- I removed a CTA that promotes the creation of extensions.  

The text is TBD (cc @jjinnii @ryankscott) and I think that we should also link the FAQs or any other public information from every notice banner to allow users to find out more.

### Docs pages

I did not test those changes since I have no idea how to build the docs locally and I couldn't find out in the one minute I was looking for it. 😐 

### Extensions header

Screenshot 2022-07-21 at 17 10 17
Screenshot 2022-07-21 at 17 10 11
Screenshot 2022-07-21 at 17 09 02

### Removed CTA

Screenshot 2022-07-21 at 17 09 27

## App preview:

- [Web](https://sg-web-ps-add-extension-deprecation.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-jyhkrkatgx.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by ryankscott** [comment]
@jjinnii  I don't think it has, I would refer to an "upcoming release"? Then in the learn more we can link to a document about it (yet to be written). My intent is to move our internal FAQ to an external page about the deprecation.

What do you think?

**Comment by jjinnii** [comment]
Thanks @ryankscott! Just to be crystal clear on the correct steps here, as we need to act quickly on this from my understanding:
1. @ryankscott To publish external FAQ page about deprecation
2. @philipp-spiess I will provide design (guidance) on a homepage/main navigation alert by Monday (have asked the Cloud team how/where they communicated that we're no longer supporting Cloud), as this is a breaking change to our product.
3. Plan on Mon/Tues shipping the extension page banners along with the home page alert. 

Does this sound about right to everyone?

**Comment by philipp-spiess** [comment]
Should we use the same timeline for the docs changes?

**Comment by jjinnii** [comment]
@philipp-spiess By docs changes, do you mean we should ship these banners whe[n we've removed the docs pages on the extensions + registry](https://docs.google.com/document/d/1wZqz7C1u6bnxYyEktILt4jJyLLkSBa8DIJLO_cvxpUM/edit#heading=h.1zx7m85kpvk9)? Sorry not sure what you mean here.

**Comment by philipp-spiess** [comment]
@jjinnii Oh, sorry for the confusion.

The screenshots I've added are from the home page and are specifically useful on the extension registry of sourcegraph.com.

There's another part in this PR that adds a warning message to all of our documentation pages that mention extensions, e.g. this page: https://docs.sourcegraph.com/extensions

Do we want to ship all of these at the same time?

---

I also just spotted another area that we might want to change and that is our marketing website. Extensions are mentioned here and the link goes to the extension registry: https://about.sourcegraph.com/code-search

We should change this to focus on IDE integrations IMO. What do you think?

**Comment by jjinnii** [comment]
@philipp-spiess  I think any part of our product (core or user docs) that mention deprecation of extensions should go out together so it’s a polished and cohesive message! As this is a major change that could be received negatively, we wouldn’t want for example, users to see this banner in our user docs but not see the same information on Sourcegraph.com

I’m not super concerned about the marketing page, as once it links to the extension registry they’ll see the banner. But maybe that’s poor product marketing to lead users to a deprecated product area. I’ll defer to @iskyOS @ryankscott on this specific decision.

**Comment by philipp-spiess** [comment]
@jjinnii Sorry I overlooked your earlier questions :( 

> Having trouble trying to find the padding below the banner. Does our banner component add padding by default?

Yes the banner adds 1rem margin bottom and the content adds another 1rem padding top.
Screenshot 2022-07-22 at 14 28 38

I've added 1.5rem margin top now between the title and the banner to try to offset it but it was rather random.

> This can be done in another issue, but we likely need to add a main navigation level alert to make sure that users are properly informed, as these banners require that users are visiting our extensions pages. I've reached out to the Cloud team to ask what alerts they utilized when we were removing our Cloud offering. I can create a follow-up issue once I have design guidance here.

Agree that this top level notice makes sense. Would you do this _in addition_ to the deprecation notices on the extensions pages? 

My thinking right now is that we need both because the top level banner will probably be dismissible.

**Comment by ryankscott** [comment]
Just an update from me. We're looking at this communication a bit more holistically and so we'll see where these deprecation messages fit with the rest of the messaging. You can follow along in this [doc](https://docs.sourcegraph.com/dev/background-information/architecture/sourcegraph-extensions).

**Comment by philipp-spiess** [comment]
@ryankscott I think you posted the wrong link?

**Comment by ryankscott** [comment]
> @ryankscott I think you posted the wrong link?

Ah thanks 🤦 , updated.

**Comment by philipp-spiess** [comment]
I've rebased the changes against the latest `main` changes. Since we're still blocked by this from design and product, I may not get to wrap this up before my PTO. @taras-yemets If we're unblocked next week when I am out, would you be able to take over this work? It should be mostly wording changes and perhaps the additional use of a site wide banner that.

**Comment by ryankscott** [comment]
@philipp-spiess  / @taras-yemets the agreed upon docs content is available [here](https://docs.google.com/document/d/1fj3upobvrLM-Mzu5IXTHpcIbCG3KrTCC_Ee6URCdF3U/edit#bookmark=id.1ovh3s6mi7dr). I think we'll need a `deprecation.md` page at the /docs/extensions/ directory and the "Learn More" links to be redirected there. 

What's the easiest way of getting this incorporated in this PR?

**Comment by taras-yemets** [comment]
@ryankscott,

 > I think we'll need a deprecation.md page at the /docs/extensions/ directory and the "Learn More" links to be redirected there.

Done in https://github.com/sourcegraph/sourcegraph/pull/39238/commits/d2664a82f969472cca2c5c8705bd643571b474f1
Screenshot 2022-08-04 at 13 07 23

**Comment by taras-yemets** [comment]
@ryankscott, there's a link on the extensions docs page that no longer leads to the extensions page on spourcegraph.com (it no longer exists). Should we remove it (and similar ones)?
Screenshot 2022-08-04 at 13 04 54

**Comment by ryankscott** [comment]
> @ryankscott, there's a link on the extensions docs page that no longer leads to the extensions page on spourcegraph.com (it no longer exists). Should we remove it (and similar ones)? 

@taras-yemets Yes we should. In fact I wonder if we should do this before the other docs changes as these links are already broken.

**Comment by taras-yemets** [comment]
>@taras-yemets Yes we should. In fact I wonder if we should do this before the other docs changes as these links are already broken.

Makes sense to me 👍🏻  Will create a separate PR for it.

**Comment by ryankscott** [comment]
@taras-yemets  / @philipp-spiess  we've just updated our comms plan and we should merge this on Monday 8th.

**Comment by ryankscott** [comment]
FYI @taras-yemets / @philipp-spiess - we are still making changes to the docs content. We should ensure we take the latest content from the Google Doc before releasing.

**Comment by taras-yemets** [comment]
> we are still making changes to the docs content. We should ensure we take the latest content from the Google Doc before releasing

@ryankscott, thank you! I'll update the deprecation page on Monday the 8th before merging.

**Comment by philipp-spiess** [comment]
@ryankscott Updated the docs. Are we okay with the wording on the notice banners:

image

And the docs:

```
> NOTE: Sourcegraph extensions are being deprecated with the upcoming Sourcegraph 4.0 release. [Learn more](../../../extensions/deprecation.md).
```

File: doc/extensions/authoring/ux_style_guide.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Extension UX style guide
 
+> NOTE: Sourcegraph extensions are being deprecated with the upcoming Sourcegraph 4.0 release. [Learn more](../deprecation.md).
```
Comment (ryankscott) [inline doc/extensions/authoring/ux_style_guide.md:3]: Need to refer to the "September release" not 4.0

File: doc/extensions/index.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 # Sourcegraph extensions
+
+> NOTE: Sourcegraph extensions are being deprecated with the upcoming Sourcegraph 4.0 release. [Learn more](./deprecation.md).
```
Comment (ryankscott) [inline doc/extensions/index.md:3]: Need to refer to the "September release" not 4.0

File: doc/extensions/principles.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Principles of extensibility for Sourcegraph
 
+> NOTE: Sourcegraph extensions are being deprecated with the upcoming Sourcegraph 4.0 release. [Learn more](./deprecation.md).
```
Comment (ryankscott) [inline doc/extensions/principles.md:3]: Need to refer to the "September release" not 4.0

File: doc/extensions/security.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # Security and privacy of Sourcegraph extensions
 
+> NOTE: Sourcegraph extensions are being deprecated with the upcoming Sourcegraph 4.0 release. [Learn more](./deprecation.md).
```
Comment (ryankscott) [inline doc/extensions/security.md:3]: Need to refer to the "September release" not 4.0

File: doc/extensions/usage.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # Using Sourcegraph extensions
 
+> NOTE: Sourcegraph extensions are being deprecated with the upcoming Sourcegraph 4.0 release. [Learn more](./deprecation.md).
```
Comment (ryankscott) [inline doc/extensions/usage.md:3]: Need to refer to the "September release" not 4.0

**Pull request merged**